src/eventer/eventer_epoll_impl.c

```diff
@@ -324,6 +324,7 @@ static void eventer_epoll_impl_trigger(eventer_t e, int mask) {
       eventer_deref(e);
       return;
     }
+    if(master_fds[fd].e == NULL) { 
```

Comment:
Line 312 just checked this above.  Won't this always be false?  If not, this likely needs a comment explaining why not after only a single if-branch has executed.

Reply:
No. we have the lock now, it could have changed in between.

Reply:
Sorry, indentation got me.